### PR TITLE
Migration and deploy

### PR DESCRIPTION
**Pull request by nbailo**
HP-32, HP-36, HP-39

### CODE REVIEW

File: contracts/VestingSwap.sol [hunk 1]
```diff
@@ -40,39 +41,37 @@ contract VestingSwap is Ownable {
...
         alphaPresale = _alphaPresale;
         betaPresale = _betaPresale;
         gammaPresale = _gammaPresale;
 
-        swap[alphaPresale].sold = IPresale(alphaPresale).totalSold();
-        swap[betaPresale].sold = IPresale(betaPresale).totalSold();
-        swap[gammaPresale].sold = IPresale(gammaPresale).totalSold();
-
-        _initVestingData();
+        // _initVestingData();
```
Comment (a-melnichuk) [inline contracts/VestingSwap.sol]: Remove commented out code.

File: scripts/deploy.ts [hunk 2]
```diff
@@ -1,18 +1,94 @@
...
-  const contractDetails = new ContractDetails();
-  const mintable = await new MintableErc20Factory(account).deploy(
-    contractDetails.name,
-    contractDetails.symbol,
-    contractDetails.totalSupply,
-    contractDetails.decimals,
-    contractDetails.maxMintedSupply,
+  const settings = new Settings();
+
+  const [deployer] = await hre.ethers.getSigners();
```
Comment (a-melnichuk) [inline scripts/deploy.ts]: Why use `hre.ethers` when you import `ethers`?

Comment (a-melnichuk) [inline scripts/deploy.ts]: Can `ethers` be used instead of `hre.ethers`?

File: test/presale/presale.ts [hunk 3]
```diff
@@ -181,48 +182,51 @@ async function testPresaleContracts(
       );
     });
 
-    it('fail - invalid sendERC20 (SafeTransfer: SEND_ERC20)', async () => {
```
Comment (a-melnichuk) [inline test/presale/presale.ts:184]: Please explain, why is this pull commented out?

**Review by a-melnichuk — CHANGES_REQUESTED** [review]

File: test/presale/presale.ts (hunk 3 above)
Comment (nbailo) [reply test/presale/presale.ts:184]: It's temporary, this test broke all process after it during execution, solving

File: contracts/VestingSwap.sol [hunk 4]
```diff
@@ -31,6 +45,9 @@ contract VestingSwap is Ownable {
     mapping(address => SwapInfo) public swap;
     mapping(address => uint) public swappedAmountOf;
 
+    event AlphaSwapInitialized();
```
Comment (a-melnichuk) [inline contracts/VestingSwap.sol]: Events are designed for client application. Empty events are not needed. Pass `start` and `percentages`.

**Review by a-melnichuk — CHANGES_REQUESTED** [review]

**Review by Aksonov-Artem — APPROVED** [review]

**Review by a-melnichuk — CHANGES_REQUESTED** [review]
Use this license line everywhere:
//SPDX-License-Identifier: Apache-2.0

**Pull request merged**